### ISSUE TITLE
`Type[Any]` is incorrectly interpreted as `Any`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2680]
This was pointed out in https://github.com/python/mypy/issues/11684.

```python
from typing import Type, Any
from typing_extensions import TypedDict

def is_type(x: object, y: Type[Any]) -> bool:
    return isinstance(x, y)

class TD(TypedDict):
    a: int

is_type(1, int)
is_type(1, TD)
is_type(1, 1)  # This should generate an error
```

**Comment by JelleZijlstra** [comment]
According to PEP 484 `Type[Any]` should be equivalent to plain `type`:

> When Type is parameterized it requires exactly one parameter. Plain Type without brackets is equivalent to Type[Any] and this in turn is equivalent to type (the root of Python's metaclass hierarchy).

But pyright does produce an error for this example when I put `type` in the signature for `is_type()`.

**Comment by erictraut** [comment]
Yes, it should treat `Type[Any]` as equivalent to `type`. That's my planned fix.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.194, which I just published. It will also be included in the next release of pylance.
